src/main/java/uk/gov/justice/hmpps/offenderevents/services/EventRetrievalService.java

```diff
@@ -27,44 +27,51 @@
     private final PollAuditRepository repository;
     private final int pollInterval;
     private final int maxEventRangeHours;
+    private final int windBackSeconds;
 
 
     public EventRetrievalService(final ExternalApiService externalApiService,
                                  final SnsService snsService,
                                  final PollAuditRepository repository,
                                  @Value("${application.events.poll.frequency:60000}") final int pollInterval,
+                                 @Value("${wind.back.seconds:10}") int windBackSeconds,
                                  @Value("${application.events.max.range.hours:1}") final int maxEventRangeHours) {
         this.externalApiService = externalApiService;
         this.snsService = snsService;
         this.repository = repository;
         this.pollInterval = pollInterval;
         this.maxEventRangeHours = maxEventRangeHours;
+        this.windBackSeconds = windBackSeconds;
+        log.info("Using {} wind back seconds", windBackSeconds);
     }
 
-    public void pollEvents() {
+    public void pollEvents(final LocalDateTime currentTime) {
 
-        final var currentTime = now();
```

Comment:
I did struggle understanding this at first; maybe consider renaming some variables;

e.g I think `final var events = externalApiService.getEvents(audit.getNextRunTime(), endTime);`

would read better if it was `final var events = externalApiService.getEvents(startTime, endTime);` so maybe introduce a variable `startTime` and possible extracting a method that adjustment you need to make for with it exceeds the max time

Reply:
agreed, I thought the same!